benchexec/tools/fdse.py

```diff
@@ -0,0 +1,50 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import benchexec.tools.template
+import benchexec.result as result
+
+
+class Tool(benchexec.tools.template.BaseTool2):
+    """
+    Tool info for FDSE.
+    https://github.com/passengerZ/FDSE
+    """
+
+    def executable(self, tool_locator):
+        return tool_locator.find_executable("fdse")
+
+    def version(self, executable):
+        return self._version_from_tool(executable, "-v")
+
+    def cmdline(self, executable, options, task, rlimits):
+        new_option = []
+        if task.property_file:
+            new_option += [f"--property-file={task.property_file}"]
+        if rlimits.cputime:
+            new_option += [f"-mt={rlimits.cputime}"]
+
+        new_option += [f"-sf={task.single_input_file}"]
+        return [executable] + new_option + options
+
+    def name(self):
+        return "FDSE"
+
+    def determine_result(self, run):
+        for line in run.output:
+            if "[BUG]" in line:
+                if "ASSERTION FAIL!" in line:
+                    return result.RESULT_FALSE_REACH
+                elif "Index out-of-bound" in line:
+                    return result.RESULT_FALSE_DEREF
+                elif "overflow" in line:
+                    return result.RESULT_FALSE_OVERFLOW
+                else:
+                    return result.RESULT_ERROR
+            if "Done : End analysis" in line:
+                return result.RESULT_DONE
+        return result.RESULT_UNKNOWN
```

Comment:
If your tool does not manage to produce any of the above output, does this really indicate "unknown" (i.e., tool gave up). Wouldn't it be an error? (Consider cases like crashes, seg faults, OOM, etc.)

Reply:
Hello, my tool is currently only intended to participate in the coverage of the testcomp, so I have removed the output classification.

Reply:
The capture error is currently based on the signal provided by the system, so it has not yet been able to accurately distinguish the error.

Reply:
Sorry, I don't really understand what you mean with "capture error". If you want to get the exit value of your program, you can do so via `run.exit_code`.

If you just want to distinguish between "done" and "error", this is also fine. Then just please change `RESULT_UNKNOWN` to `RESULT_ERROR` in order to do so.

Reply:
Thank you for your suggestion. Now the output can be divided into three types: done, timeout, and error.